src/test/unit_tests/jest_tests/test_config_dir_restructure_upgrade_script.test.js

```diff
@@ -0,0 +1,654 @@
+// which is called when using noobaa-cli, for having an accurate test, it'll be blocked from running on an 
+// env having an existing default config directory and the test executer will be asked to remove the default 
+// config directory before running the test again, do not run on production env or on any env where the existing config directory is being used
+describe('move_old_accounts_dir', () => {
+
+    beforeAll(async () => {
+        await fail_test_if_default_config_dir_exists('test_config_dir_restructure', default_config_fs);
+        await create_config_dir(config.NSFS_NC_DEFAULT_CONF_DIR);
+    });
+
+    afterEach(async () => {
+        await clean_config_dir(default_config_fs);
+    }, TEST_TIMEOUT);
+
+    it('move_old_accounts_dir() - accounts/ dir is missing', async () => {
+        await move_old_accounts_dir(default_config_fs, [], mock_old_version, dbg);
+        await assert_old_account_dir_was_deleted();
+        await assert_backup_dir({});
+    });
+
+    it('move_old_accounts_dir() - accounts/ dir is empty', async () => {
+        await create_fresh_path(default_config_fs.old_accounts_dir_path);
+        await move_old_accounts_dir(default_config_fs, [], mock_old_version, dbg);
+        await assert_old_account_dir_was_deleted();
+        await assert_backup_dir({});
+    });
+
+    it('move_old_accounts_dir() - accounts/ dir contains a single account', async () => {
+        await create_fresh_path(default_config_fs.old_accounts_dir_path);
+        const account_name = 'old_account1';
+        const new_account_data = { _id: '1', name: account_name, user: 'root' };
+        await write_manual_old_account_config_file(default_config_fs, new_account_data);
+        await move_old_accounts_dir(default_config_fs, [account_name], mock_old_version, dbg);
+        await assert_old_account_dir_was_deleted();
+        await assert_backup_dir({ [account_name]: new_account_data});
+    });
+
+    it('move_old_accounts_dir() - accounts/ dir contains 5000 accounts', async () => {
+        await create_fresh_path(default_config_fs.old_accounts_dir_path);
+        const account_ids = Array.from({ length: 5000 }, (_, i) => Number(i + 1));
+        const account_names_obj = {};
+        await P.map_with_concurrency(5000, account_ids, async account_id => {
+            const new_account_data = { _id: String(account_id), name: 'old_account' + account_id, user: 'root' };
+            await write_manual_old_account_config_file(default_config_fs, new_account_data);
+            account_names_obj[new_account_data.name] = new_account_data;
+        });
+        await move_old_accounts_dir(default_config_fs, Object.keys(account_names_obj), mock_old_version, dbg);
+        await assert_old_account_dir_was_deleted();
+        await assert_backup_dir(account_names_obj);
+    }, TEST_TIMEOUT);
+
+    it('move_old_accounts_dir() - .backup_accounts_dir already exists and contains some of the accounts in accounts/ dir', async () => {
```

Comment:
Why "some of the accounts in accounts/ dir"?
In `await assert_backup_dir(account_names_obj);` the `account_names_obj` contains all the accounts, no?

Reply:
I meant that some of the accounts are in the backup dir, thanks!